src/chat.cpp

```diff
@@ -731,6 +738,11 @@ void ChatBackend::clearRecentChat()
 	m_recent_buffer.clear();
 }
 
+
+void ChatBackend::applySettings(Settings* settings) {
+	m_recent_buffer.resize(settings->getU32("recent_chat_size"));
```

Comment:
Why are you passing g_settings to this function instead of accessing it directly?